Neos.Cache/Classes/Backend/RedisBackend.php

```diff
@@ -217,33 +227,26 @@ public function remove(string $entryIdentifier): bool
      * in an atomic way.
      *
      * @throws \RuntimeException
-     * @return void
      * @api
      */
     public function flush(): void
     {
+        // language=lua
         $script = "
-		local entries = redis.call('LRANGE',KEYS[1],0,-1)
-		for k1,entryIdentifier in ipairs(entries) do
-			redis.call('DEL', ARGV[1]..'entry:'..entryIdentifier)
-			local tags = redis.call('SMEMBERS', ARGV[1]..'tags:'..entryIdentifier)
-			for k2,tagName in ipairs(tags) do
-				redis.call('DEL', ARGV[1]..'tag:'..tagName)
-			end
-			redis.call('DEL', ARGV[1]..'tags:'..entryIdentifier)
-		end
-		redis.call('DEL', KEYS[1])
-		redis.call('DEL', KEYS[2])
-		";
-        $this->redis->eval($script, [$this->getPrefixedIdentifier('entries'), $this->getPrefixedIdentifier('frozen'), $this->getPrefixedIdentifier('')], 2);
+        local keys = redis.call('KEYS', ARGV[1] .. '*')
+        for k1,key in ipairs(keys) do
+            redis.call('DEL', key)
+        end
+        ";
+        $this->redis->eval($script, [$this->getPrefixedIdentifier('frozen'), $this->getPrefixedIdentifier('')], 1);
```

Comment:
`$this->getPrefixedIdentifier('frozen')` is not needed anymore, is it? At least KEYS[1] is never accessed.

Reply:
It is still needed, but properly the script is incorrect now. Will check and adjust the test. Thx for noticing!

Reply:
Ok, you are right, the key is already included in the args with wildcard.